src/irmin/tree.ml

```diff
@@ -892,7 +972,7 @@ module Make (P : Backend.S) = struct
       | None -> t.info.findv_cache <- Some (StepMap.singleton step v)
       | Some m -> t.info.findv_cache <- Some (StepMap.add step v m)
 
-    let findv ~cache ctx t step =
+    let findv_aux ~cache ~value_of_key ~return ~bind ctx t step =
```

Comment:
What is the reason to introduce a `return` and a `bind` parameter in this function? I'm not sure to understand their purpose...

Reply:
Because it's used in `Proof.findv` which isn't in the Lwt monad.

Reply:
Oh, I see! Thanks for the explanation :+1: